ops/gauntlet/gauntlet_plus_plus_starknet.go

```diff
@@ -0,0 +1,313 @@
+package gauntlet
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"context"
+	"github.com/rs/zerolog/log"
+	g "github.com/smartcontractkit/gauntlet-plus-plus/sdks/go-gauntlet/client"
+)
+
+var (
+	sgpp *StarknetGauntletPlusPlus
+)
+
+type Request struct {
+	Input map[string]any 		 `json:"input"`
+	Command string					 `json:"command"`
+}
+
+type StarknetGauntletPlusPlus struct {
+	client				*g.ClientWithResponses
+	gr      			*http.Response
+	providers 		*[]g.Provider
+}
+
+func (sgpp *StarknetGauntletPlusPlus) BuildProviders(address string, rpcUrl string, privateKey string) (*[]g.Provider) {
+	var input map[string]*interface{}
+	input = make(map[string]*interface{})
+	addressValue := interface{}(address)
+	input["address"] = &addressValue
+	AccountProvider := g.Provider{
+		Name: "basic-address",
+		Type: "@chainlink/gauntlet-starknet/lib/starknet.js/account",
+		Input: input,
+	}
+
+	input = make(map[string]*interface{})
+	privateKeyValue := interface{}(privateKey)
+	debugValue := interface{}(true)
+	input["privateKey"] = &privateKeyValue
+	input["debug"] = &debugValue
+	SignerProvider := g.Provider{
+		Name: "basic-pk",
+		Type: "@chainlink/gauntlet-starknet/lib/starknet.js/signer",
+		Input: input,
+	}
+
+	input = make(map[string]*interface{})
+	rpcUrlValue := interface{}(rpcUrl)
+	checkStatusValue := interface{}(true)
+	input["url"] = &rpcUrlValue
+	input["checkStatus"] = &checkStatusValue
+	RpcProvider := g.Provider{
+		Name: "basic-url",
+		Type: "@chainlink/gauntlet-starknet/lib/starknet.js/provider",
+		Input: input,
+	}
+
+	providers := []g.Provider{AccountProvider, SignerProvider, RpcProvider}
+
+	return &providers
+}
+
+// New StarknetGauntletPlusPlus creates a default g++ client with responses
+func NewStarknetGauntletPlusPlus(gauntletPPEndpoint string, rpcUrl string, address string, privateKey string) (*StarknetGauntletPlusPlus, error) {
+	fmt.Println("rpcUrl: " + rpcUrl)
+	fmt.Println("GPP URL: " + gauntletPPEndpoint)
+	newClient, err := g.NewClientWithResponses(gauntletPPEndpoint)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sgpp = &StarknetGauntletPlusPlus{
+		client: newClient,
+		gr: &http.Response{},
+		providers: sgpp.BuildProviders(address, rpcUrl, privateKey),
+	}
+
+
+	return sgpp, nil
+}
+
+func (sgpp *StarknetGauntletPlusPlus) ExtractValueFromResponseBody(report g.Report, key string) string {
+	if report.Output != nil {
+		// Attempt to assert the Output as a map
+		if outputMap, ok := (*report.Output).(map[string]interface{}); ok {
+			if value, exists := outputMap[key]; exists {
+				// Assert value to a string
+				if strValue, ok := value.(string); ok {
+					fmt.Println("Value:", strValue)
+					return strValue
+				} else {
+					fmt.Println("Value is not of type string")
```

Comment:
instead of printing out the issue, you can return an error here

so you'll change your return type of the method to `string, error` 

and in the function that calls ExtractValueFromResponseBody you check if there is an error. If so, you can log the error and either bubble the error up or just do nothing depending on what behavior you want. In this case, I think we should probably bubble the error up since we expect to get a value

Reply:
Added better error catching